### ISSUE TITLE
4.0alpha: Stretch scale factor doesn't apply when using `viewport` stretch mode (while it does in `3.x`)

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot#60801]
### Godot version

4.0.alpha7

### System information

Fedora 34, GeForce GTX 1080 (NVIDIA 510.60.02)

### Issue description

In `master`, the stretch scale factor doesn't apply when using the `viewport` stretch mode (while it does in 3.5beta4). The stretch scale factor works as expected when using the `disabled` or `canvas_items` stretch modes.

cc @Ansraer, as they ported the stretch scale to 4.0 inhttps://github.com/godotengine/godot/pull/52170.

### `master` behavior (incorrect)

https://user-images.githubusercontent.com/180032/166980544-a9652520-f5cb-4fb3-829a-e9a2b708666d.mp4

### `3.x` behavior (correct)

https://user-images.githubusercontent.com/180032/166980533-b8139c60-2635-4b09-97f4-137fbb094f65.mp4

### Steps to reproduce

- Use the `viewport` stretch mode (not `disabled` or `canvas_items`).

### Minimal reproduction project

- `master` project: https://github.com/Calinou/godot-demo-projects/tree/multiple-resolutions-update-to-4.0/gui/multiple_resolutions
- `3.x` project: https://github.com/godotengine/godot-demo-projects/tree/master/gui/multiple_resolutions

**Comment by Ansraer** [comment]
I am fairly certain it isn't supposed to work for viewport in 3.x. When I wrote the original PR there was a reason why I excluded that (see the PR messages), though I can't remember what it was right now.
What makes this even more confusing is that, despite what I wrote in the messages, according to git blame I did add code for viewport stretching in 3.x.

I suppose I could enable this in 4.x and see what happens.
